docs/src/design/range.md

```diff
@@ -103,102 +103,66 @@ This 3-column table addresses the limitations we had as follows:
+This allows us to reduce the minimum trace length to 64.
 
-## Optimizations
+To find out the number of bridge rows to be added in between two values to be range checked, we represent the gap between them as a linear combination of powers of 3, ie,
 
-First, we can just stack the tables on top of each other. We'll need to add a column to partition the table between the sections used for 8-bit range checks and sections used for 16-bit range checks. Let's call this column $t$. When $t = 0$, we'll apply constraints for the 8-bit table, and when $t = 1$, we'll apply constraints for the 16-bit table.
-
-![rc_optimisation.png](../assets/design/range/rc_optimisation.png)
-
-Second, we can merge running product columns $p_0$ and $p_1$ into a single column. We'll do it like so:
-
-- When $t = 0$, we'll multiply the current value of the column by the 8-bit value (offset by random $\alpha_0$) that we want to add into the running product.
-- When $t = 1$, we'll divide the current value of the column by the 8-bit value (offset by random $\alpha_0$) which we'd like to remove from the running product.
-
-In the end, if we added and then removed all the same values, the value in this column (let's call it $p_0$) should equal to $1$, and we can check this condition via a boundary constraint.
+$$
+(r' - r) = \sum_{i=0}^{7} x_i \cdot 3^i
+$$
 
-The only downside of this construction is again higher constraint degree. Specifically, some of the transition constraints described above have degree $8$. However, this doesn't matter in the context of Miden VM, since the max constraint degree of the VM is $9$.
+Then for each $x_i$ except the first, we add a bridge row at a gap of $3^i$.
 
 ## Miden approach
 
-This final optimized construction is implemented in Miden with the following requirements, capabilities, and constraints.
+This construction is implemented in Miden with the following requirements and constraints.
 
 ### Requirements
 
-- 4 columns of the main trace: $t, s_0, s_1, v$.
+- 3 columns of the main trace: $s_0, s_1, v$.
 - 1 [bus](./multiset.md#communication-buses) $b_{range}$ to ensure that the range checks performed in the range checker match those requested by other VM components (the [stack](./stack/u32_ops.md#range-checks) and the [memory chiplet](./chiplets/memory.md)).
-- 1 [virtual table](./multiset.md#virtual-tables), tracked in running product column $p_0$, which enables skipping values by up to $255$ in the 16-bit section of the trace and enforces consistency between the range checks in the 8-bit section of the trace and the value increments in the 16-bit section of the trace.
-
-### Capabilities
 
-The construction gives us the following capabilities:
-- For long traces (when $n > 2^{16}$), we can do over $3n$ arbitrary 16-bit range-checks.
-- For short traces ($2^{10} < n \le 2^{16}$), we can range-check at slightly fewer than $n$ unique values, but if there are duplicates, we may be able to range-check up to $3n$ total values.
 
```

Comment:
@bobbinth I'm not sure how these were calculated and how these should be adjusted for the new construct.

Reply:
I think this could be updated as follows:

- For long traces (when $n > 2^{16}$), we can do over $4n$ arbitrary 16-bit range-checks.
- For short traces ($2^5 < n \le 2^{16}$), we can range-check at slightly fewer than $n$ unique values, but if there are duplicates, we may be able to range-check up to $4n$ total values.

Reply:
Thank you.